pkg/azuredisk/controllerserver.go

```diff
@@ -476,9 +474,16 @@ func (d *Driver) ControllerPublishVolume(ctx context.Context, req *csi.Controlle
 		klog.V(2).Infof("attach volume %q to node %q successfully", diskURI, nodeName)
 	}
 
-	pvInfo := map[string]string{consts.LUN: strconv.Itoa(int(lun))}
+	publishContext := map[string]string{consts.LUN: strconv.Itoa(int(lun))}
+	volumeConext := req.VolumeContext
+	if disk != nil && volumeConext != nil {
+		if _, ok := volumeConext[consts.RequestedSizeGib]; !ok {
```

Comment:
Is it necessary to restrict this to static provisioned PVs? In the dynamic case, the metadata could be redundant to that already in the PV but clients would take the same approach to get it (by always looking at the `VolumeAttachment`). This would avoid the need for having two different ways of getting the metadata in the dynamic and static cases.

Reply:
As discussed, for dynamic PVs, azure back service should use metadata in PV which is more reliable and generic. Since exposing disk metadata in `volumeattachments` is only required by azure back service, I will leave this question to @shrac

Reply:
We want to read VolumeAttachment objects only where required (for now static PVs) as that would mean fetching more info and might delay the validation call - Also, I could not find a way to filter a VolumeAttachment by its PV name/uuid. 

Also, there is no guarantee that VolumeAttachment would always exist. Ex: For PVs that were never attached (or) existing PVs which are already attached and the new node publish call is yet to come. As the number of such static PVs should be less we will fail validation for now but we cannot do the same for dynamic PVs.  Hence, we would continue relying on PV volume attributes for dynamic PVs.

Reply:
Right. Makes sense.